### PR TITLE
[Chat] GA AttachmentMetadata api

### PR DESCRIPTION
**Pull request by jimchou-dev**
# What

- GA AttachmentMetadata api

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by Leah-Xia-Microsoft — APPROVED** [review]

**Review by palatter — APPROVED** [review]

File: packages/acs-ui-common/src/common.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export function _getKeys<T extends object>(obj: T): (keyof T)[] {
  * a name that represents the name of file, and
  * a URL to download the attachment.
  *
- * @beta
+ * @public
```
Comment (emlynmac) [inline packages/acs-ui-common/src/common.ts:55]: There should be changes to the stable API .md, is there a feature flag that needs to be added to the stabilized section?

Comment (jimchou-dev) [reply packages/acs-ui-common/src/common.ts:55]: remove the cc flag in index.js to separate export, now it shows up, good catch thank you!

File: packages/communication-react/review/stable/communication-react.api.md [hunk 2]
```diff
@@ -1555,6 +1562,7 @@ export type ChatInitializedListener = (event: {
 export interface ChatMessage extends MessageCommon {
     // (undocumented)
     attached?: MessageAttachedStatus;
+    attachments?: AttachmentMetadata[];
```
Comment (emlynmac) [inline packages/communication-react/review/stable/communication-react.api.md:1565]: I'm unclear as to why we're stabilizing this specific property - it doesn't seem to add any stabilized feature?

**Review by emlynmac — CHANGES_REQUESTED** [review]
We need to update this to have more functionality

File: packages/communication-react/review/stable/communication-react.api.md [hunk 3]
```diff
@@ -1440,7 +1473,7 @@ export type ChatAdapterUiState = {
 };
 
 // @public
-export type ChatAttachmentType = 'unknown' | 'image';
+export type ChatAttachmentType = 'unknown' | 'image' | /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */ 'file';
```
Comment (emlynmac) [inline packages/communication-react/review/stable/communication-react.api.md:1476]: This seems like it shouldn't be here?

File: packages/communication-react/review/stable/communication-react.api.md [hunk 4]
```diff
@@ -1498,11 +1531,13 @@ export type ChatCompositeOptions = {
     errorBar?: boolean;
     topic?: boolean;
     autoFocus?: 'sendBoxTextField';
+    attachmentOptions?: AttachmentOptions;
 };
 
 // @public
 export interface ChatCompositeProps extends BaseCompositeProps<ChatCompositeIcons> {
     adapter: ChatAdapter;
+    formFactor?: 'desktop' | 'mobile';
```
Comment (emlynmac) [inline packages/communication-react/review/stable/communication-react.api.md:1540]: Where did this sneak in from? Is it part of this?

File: packages/communication-react/review/stable/communication-react.api.md (hunk 3 above)
Comment (jimchou-dev) [reply packages/communication-react/review/stable/communication-react.api.md:1476]: it's still under CC flag, similar to this
`// @public
export type CallAdapterLocator = TeamsMeetingLinkLocator | GroupCallLocator | RoomCallLocator | /* @conditional-compile-remove(meeting-id) */ TeamsMeetingIdLocator;`

Comment (emlynmac) [reply packages/communication-react/review/stable/communication-react.api.md:1476]: This should be stripped - i.e. no comment in the API flag. It's a bit strange to have comments like this in the TypeScript definitions publicly.
The stable API should only have stabilized features, which in this case includes the 'file' addition. It looks like the CC step is not stripping the comments out in this case.

Comment (jimchou-dev) [reply packages/communication-react/review/stable/communication-react.api.md:1476]: Craeted 3745338 to track the issue of cc comments to be fixed

**Pull request closed without merging**